core/templates/domain/exploration/ExplorationDraftObjectFactory.ts

```diff
@@ -20,24 +20,47 @@
 import { downgradeInjectable } from '@angular/upgrade/static';
 import { Injectable } from '@angular/core';
 
+export interface IAddStateNameChangeList {
+  'cmd': string;
+  'state_name': string;
+}
+
+export interface IEditExplorationPropertyChangeList {
+  'cmd': string;
+  'new_value': Object;
+  'old_value': Object;
+  'property_name': string;
+}
+
+export interface IEditStatePropertyChangeList {
+  'cmd': string;
+  'new_value': Object;
+  'old_value': Object;
+  'property_name': string;
+  'state_name': string;
+}
+
+export interface IRenameStateChangeList {
+  'cmd': string;
+  'new_state_name': string;
+  'old_state_name': string;
+}
+
+export type IExplorationChangeList = (
+  IAddStateNameChangeList | IEditExplorationPropertyChangeList |
+  IEditStatePropertyChangeList | IRenameStateChangeList);
```

Comment:
Can you we use discriminated unions here? As Brian suggested in #9450.

Reply:
done